### DISCUSSION TITLE
Is there an example available for creating a context menu

### DISCUSSION THREAD

**Discussion by pkk-code** [progrium/darwinkit#166] [Q&A]
I would like to implement a context menu that appears when I right-click on files or directories in Finder.

**Comment by progrium** [comment]
No but I can make one, thanks for asking.

**Reply by progrium** [reply]
Following up, you don't happen to have an Objective-C or Swift example/tutorial you're referencing? I need to know more specifically what you mean. If I understand correctly, you want to *add* items to the context menu in Finder?

**Reply by pkk-code** [reply]
I was looking for a solution resembling the screenshot I've attached. The goal is to dynamically add menu items to finder based on file type using Go code that can perform actions. Unfortunately, I don't possess any Objective-C or Swift examples. While I did experiment with your package and explored the API, I couldn't manage to successfully add the menu item to Finder. It's possible that the challenge lies in my understanding of the inner workings of the process.


![image](https://github.com/progrium/macdriver/assets/37810104/500994c4-7bbb-4e28-a466-a6b63eb73992)

**Reply by progrium** [reply]
I've been putting it off because this requires another framework to add called FinderSync, but I'll let you know when that's in.
